tfjs-core/src/io/weights_loader.ts

```diff
@@ -245,3 +235,180 @@ export function weightsLoaderFactory(
     return weightsTensorMap;
   };
 }
+
+type BufferRange = {
```

Comment:
Naming: range -> chunk/shard/partition
And all related variable and function names

Reply:
Good point. That's a much better name. Fixed.